test/forge/HalmosTest.sol

```diff
@@ -0,0 +1,151 @@
+
+        uint256 lltv = svm.createUint256("lltv");
+        marketParams = MarketParams(address(loanToken), address(collateralToken), address(oracle), address(irm), lltv);
+        id = marketParams.id();
+
+        vm.assume(block.timestamp != 0);
+        vm.startPrank(OWNER);
+        morpho.enableLltv(lltv);
+        morpho.createMarket(marketParams);
+        vm.stopPrank();
+
+        vm.roll(block.number + 1);
+        vm.warp(block.timestamp + 1 * BLOCK_TIME);
+    }
+
+    // Call Morpho, assuming interacting with only the defined market for performance reasons.
+    function _callMorpho(bytes4 selector, address caller) internal {
+        vm.assume(selector != morpho.extSloads.selector);
+        vm.assume(selector != morpho.createMarket.selector);
+
+        bytes memory emptyData = hex"";
+        uint256 assets = svm.createUint256("assets");
+        uint256 shares = svm.createUint256("shares");
+        address onBehalf = svm.createAddress("onBehalf");
+        address receiver = svm.createAddress("receiver");
+
+        bytes memory args;
+
+        if (selector == morpho.supply.selector) {
+            args = abi.encode(marketParams, assets, shares, onBehalf, emptyData);
+        } else if (selector == morpho.withdraw.selector) {
+            args = abi.encode(marketParams, assets, shares, onBehalf, receiver);
+        } else if (selector == morpho.borrow.selector) {
+            args = abi.encode(marketParams, assets, shares, onBehalf, receiver);
+        } else if (selector == morpho.repay.selector) {
+            args = abi.encode(marketParams, assets, shares, onBehalf, emptyData);
+        } else if (selector == morpho.supplyCollateral.selector) {
+            args = abi.encode(marketParams, assets, onBehalf, emptyData);
+        } else if (selector == morpho.withdrawCollateral.selector) {
+            args = abi.encode(marketParams, assets, onBehalf, receiver);
+        } else if (selector == morpho.liquidate.selector) {
+            address borrower = svm.createAddress("borrower");
+            args = abi.encode(marketParams, borrower, assets, shares, emptyData);
+        } else if (selector == morpho.flashLoan.selector) {
+            uint256 rand = svm.createUint256("rand");
+            address token;
+            if (rand == 0) {
+                token = address(loanToken);
+            } else if (rand == 1) {
+                token = address(collateralToken);
+            } else {
+                ERC20Mock otherToken = new ERC20Mock();
+                token = address(otherToken);
+            }
+            args = abi.encode(marketParams, token, assets, emptyData);
+        } else if (selector == morpho.accrueInterest.selector) {
+            args = abi.encode(marketParams);
+        } else if (selector == morpho.setFee.selector) {
+            uint256 newFee = svm.createUint256("newFee");
+            args = abi.encode(marketParams, newFee);
+        } else {
+            args = svm.createBytes(1024, "data");
```

Comment:
This snippet of code is here for 2 reasons:
- to disambiguate dynamic types as arguments of symbolic functions, because Halmos does not support it
- to make it easier on the prover, notably here we pass exactly the marketParam for the market that was created in the setup